### PR TITLE
Migrate `lib/utils/validate*` modules to ESM

### PR DESCRIPTION
**Pull request by ybiquitous**
> Which issue, if any, is this issue related to?

Ref #5291

> Is there anything in the PR that needs further explanation?

In addition to rewriting some utility modules to ESM, this change also fine-tunes Rollup to prevent inlined modules.

See the `output.preserveModules` option for RollUp: https://rollupjs.org/configuration-options/#output-preservemodules

### CODE REVIEW

File: rollup.config.mjs [hunk 1]
```diff
@@ -16,14 +18,20 @@ for (const input of inputFiles) {
...
 export default inputFiles.map((input) => {
 	return {
 		input,
 		output: {
 			format: 'cjs',
-			file: input.replace('.mjs', '.cjs'),
+			dir: rootDir,
+			entryFileNames: ({ facadeModuleId }) => {
+				return relative(rootDir, facadeModuleId).replace('.mjs', '.cjs');
+			},
```
Comment (ybiquitous) [inline rollup.config.mjs:31]: [note] When enabling `preserveModules`, `dir` is needed, and then `entryFileNames` is also needed to output a file ending with `.cjs`.

File: lib/utils/isOnlyWhitespace.cjs [hunk 2]
```diff
@@ -1,14 +1,6 @@
...
-/**
- * Check if a character is whitespace.
- *
- * @param {string} char
- * @returns {boolean}
- */
-function isWhitespace(char) {
-	return [' ', '\n', '\t', '\r', '\f'].includes(char);
-}
+const isWhitespace = require('./isWhitespace.cjs');
```
Comment (ybiquitous) [inline lib/utils/isOnlyWhitespace.cjs:3]: [note] This is thanks to `preserveModules`.

**Review by jeddy3 — APPROVED** [review]
Nice, thank you!

**Comment by ybiquitous** [comment]
Wow, thanks for the quick review! 😄

**Pull request merged**